packages/calypso-e2e/src/lib/flows/difm-flow.ts

```diff
@@ -0,0 +1,192 @@
+	 */
+	constructor( private page: Page ) {}
+
+	/**
+	 * Given text, click on the button's first instance with the text.
+	 *
+	 * @param {string} text User-visible text on the button.
+	 */
+	async clickButton( text: string ): Promise< void > {
+		const selector = selectors.button( text );
+		await this.page.click( selector );
+	}
+
+	/**
+	 * Validates that we've are seeing the correct header.
+	 *
+	 * @param {string} title The title that the page should be showing.
+	 */
+	async validatePageTitle( title: string ): Promise< void > {
+		await this.page.waitForSelector( selectors.header( title ) );
+	}
+
+	/**
+	 * Go to first setup page.
+	 */
+	async visitSetup(): Promise< void > {
+		await this.page.goto(
+			DataHelper.getCalypsoURL(
+				'/start/do-it-for-me/new-or-existing-site?flags=signup/redesigned-difm-flow'
+			)
+		);
+	}
+
+	/**
+	 * Validates that we've landed on the starting page.
+	 */
+	async validateStartPage(): Promise< void > {
+		await this.validatePageTitle( 'Do It For Me' );
+	}
+
+	/**
+	 * Validates that we've landed on the options page.
+	 */
+	async validateOptionsPage(): Promise< void > {
+		await this.validatePageTitle( "First, let's give your site a name" );
+	}
+
+	/**
+	 * Validates that we've landed on the social page.
+	 */
+	async validateSocialPage(): Promise< void > {
+		await this.validatePageTitle( 'Do you have social media profiles?' );
+	}
+
+	/**
+	 * Validates that we've landed on the design page.
+	 */
+	async validateDesignPage(): Promise< void > {
+		await this.validatePageTitle( 'Themes' );
+	}
+
+	/**
+	 * Validates that we've landed on the page picker page.
+	 */
+	async validatePagePickerPage(): Promise< void > {
+		await this.validatePageTitle( 'Add pages to your' );
+	}
+
+	/**
+	 * Validates that we've landed on the 'choose existing' page.
+	 */
+	async validateUseExistingPage(): Promise< void > {
+		await this.validatePageTitle( 'Choose where you want us to build your site.' );
+	}
+
+	/**
+	 * Validates that the confirmation modal is being shown.
+	 */
+	async validateConfirmationBox(): Promise< void > {
+		await this.validatePageTitle( 'Site Reset Confirmation' );
+	}
+
+	/**
+	 * Validates that we've landed on the checkout page.
+	 */
+	async validateCheckoutPage(): Promise< void > {
+		await this.page.waitForSelector( selectors.checkoutHeader, { timeout: 60000 } );
+	}
+
+	/**
+	 * Enter the values for your site name as well as an optional tagline.
+	 *
+	 * @param {string} name The name for the new site.
+	 * @param {string} [tagline] An optional tagline.
+	 */
+	async enterOptions( name: string, tagline?: string ): Promise< void > {
+		await this.page.fill( selectors.siteNameInput, name );
+		if ( typeof tagline !== 'undefined' ) {
+			await this.page.fill( selectors.taglineInput, tagline );
+		}
+	}
+
+	/**
+	 * Enter the links to social media profiles.
+	 *
+	 * @param {string} twitter The link to a twitter profile.
+	 */
+	async enterSocial( twitter: string ): Promise< void > {
```

Comment:
A bit of nit: but I think this current method design may grow a little clunky? That is, if we wanted to add other social inputs in the future, we'd have to redo the signature here (and update all the corresponding places that called it).

I think two approaches that might grow a bit more smoothly would be either...
1. Just have a unique method for each form of input. So `enterTwitterAccount()`, and then we could add others in the future `enterFacebookAccount()`, etc.
2. Change the parameter to be an object with optional keys for each social media account. So something like...
```typescript
{
    twitter?: string;
    facebook?: string;
}
```
The method would then add whichever social media accounts were supplied. To start, we would only need to have the `twitter` key, but this would let us add future social media support without having to change the signature in a disruptive way!

Thoughts?

Reply:
I did not use the multiple methods option as it would in my opinion make the flow object much less readable.

Reply:
>Instead, I opted to use an interface. That way you just expand the interface and add a check inside the method. If you want to add a social media account other than Twitter, you can then add it like:

Ah yes, this is exactly what I had in mind! My apologies if I worded my comment a bit confusingly. I was imagining a single object parameter with optional properties. 👍